pkg/config/types/generated_constants.go

```diff
@@ -1,124 +1,124 @@
 const ComputeEnabledKey = "compute.enabled"
+const ComputeHeartbeatInfoUpdateIntervalKey = "compute.heartbeat.infoupdateinterval"
+const ComputeHeartbeatIntervalKey = "compute.heartbeat.interval"
+const ComputeHeartbeatResourceUpdateIntervalKey = "compute.heartbeat.resourceupdateinterval"
+const ComputeLabelsKey = "compute.labels"
 const ComputeOrchestratorsKey = "compute.orchestrators"
-const ComputeTLSCertFileKey = "compute.tls.certfile"
-const ComputeTLSKeyFileKey = "compute.tls.keyfile"
+const ComputeTLSAutoCertKey = "compute.tls.autocert"
+const ComputeTLSAutoCertCachePathKey = "compute.tls.autocertcachepath"
 const ComputeTLSCAFileKey = "compute.tls.cafile"
-const ComputeTLSUseTLSKey = "compute.tls.usetls"
+const ComputeTLSCertFileKey = "compute.tls.certfile"
 const ComputeTLSInsecureKey = "compute.tls.insecure"
+const ComputeTLSKeyFileKey = "compute.tls.keyfile"
 const ComputeTLSSelfSignedKey = "compute.tls.selfsigned"
-const ComputeTLSAutoCertKey = "compute.tls.autocert"
-const ComputeTLSAutoCertCachePathKey = "compute.tls.autocertcachepath"
-const ComputeHeartbeatInfoUpdateIntervalKey = "compute.heartbeat.infoupdateinterval"
-const ComputeHeartbeatResourceUpdateIntervalKey = "compute.heartbeat.resourceupdateinterval"
-const ComputeHeartbeatIntervalKey = "compute.heartbeat.interval"
-const ComputeLabelsKey = "compute.labels"
-const ComputeAllocatedCapacityCPUKey = "compute.allocatedcapacity.cpu"
-const ComputeAllocatedCapacityMemoryKey = "compute.allocatedcapacity.memory"
-const ComputeAllocatedCapacityDiskKey = "compute.allocatedcapacity.disk"
-const ComputeAllocatedCapacityGPUKey = "compute.allocatedcapacity.gpu"
-const ComputeAllowListedLocalPathsKey = "compute.allowlistedlocalpaths"
-const WebUIEnabledKey = "webui.enabled"
-const WebUIListenKey = "webui.listen"
+const ComputeTLSUseTLSKey = "compute.tls.usetls"
+const DataDirKey = "datadir"
+const DisableAnalyticsKey = "disableanalytics"
+const EnginesDisabledKey = "engines.disabled"
+const EnginesTypesDockerManifestCacheRefreshKey = "engines.types.docker.manifestcache.refresh"
+const EnginesTypesDockerManifestCacheSizeKey = "engines.types.docker.manifestcache.size"
+const EnginesTypesDockerManifestCacheTTLKey = "engines.types.docker.manifestcache.ttl"
+const FeatureFlagsExecTranslationKey = "featureflags.exectranslation"
 const InputSourcesDisabledKey = "inputsources.disabled"
-const InputSourcesReadTimeoutKey = "inputsources.readtimeout"
 const InputSourcesMaxRetryCountKey = "inputsources.maxretrycount"
+const InputSourcesReadTimeoutKey = "inputsources.readtimeout"
 const InputSourcesTypesIPFSEndpointKey = "inputsources.types.ipfs.endpoint"
-const InputSourcesTypesS3EndpointKey = "inputsources.types.s3.endpoint"
 const InputSourcesTypesS3AccessKeyKey = "inputsources.types.s3.accesskey"
+const InputSourcesTypesS3EndpointKey = "inputsources.types.s3.endpoint"
 const InputSourcesTypesS3SecretKeyKey = "inputsources.types.s3.secretkey"
```

Comment:
we don't support passing s3 accesskey and secretkey through the config. Please make sure only configs being used are present and listed back to the user.

Reply:
These values and their corresponding config were introduced before this change, and were a requirement from the re-thinking config document. I'd like to keep review here focused on the issue(s) being fixed. These keys are generated from our config struct, so if we want to change something here, we need to modify the config.

Reply:
they were removed in another PR. The re-thinking config document include configurations that don't exist today to showcase how the configuration can be extended in the future. You are aware of that and that is why you didn't include many other configurations such as GC related configurations. 
I understand you didn't add them in this PR, but thankfully the PR made these configurations visible and it should be fine to comment about them It doesn't mean they have to be addressed in this PR.